api/v1alpha3/common_types.go

```diff
@@ -46,6 +46,10 @@ const (
 	// that was cloned for the machine.
 	TemplateClonedFromGroupKindAnnotation = "cluster.x-k8s.io/cloned-from-groupkind"
 
+	// ClusterConfigurationAnnotation is a machine annotation that stores the json-marshalled string of KCP ClusterConfiguration.
+	// This annotation is used to detect any changes in ClusterConfiguration and trigger machine rollout in KCP.
+	ClusterConfigurationAnnotation = "cluster.x-k8s.io/cluster-configuration"
```

Comment:
```suggestion
	// KubeadmClusterConfigurationAnnotation is a machine annotation that stores the json-marshalled string of KCP ClusterConfiguration.
	// This annotation is used to detect any changes in ClusterConfiguration and trigger machine rollout in KCP.
	KubeadmClusterConfigurationAnnotation = "controlplane.cluster.x-k8s.io/cluster-configuration"
```

@detiber thoughts/preferences?

Reply:
If this annotation is only used by KCP, should we define it in the kcp types package instead?

+1 to being explicit with then naming and group, but if we do go down that route, it should probably be: `controlplane.cluster.x-k8s.io/kubeadm-cluster-configuration`